src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -1096,12 +1096,10 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
               case (Apply(Select(receiver, _), _), SingleType(_, sym)) => sym == receiver.symbol
               case _ => false
             }
-            if (!isThisTypeResult) {
-              if (tree.tpe.termSymbol == UnitClass.companionModule)
-                context.warning(tree.pos, "discarded companion object Unit of type Unit.type. Use () to obtain a value of type Unit.")
-              else if (settings.warnValueDiscard)
-                context.warning(tree.pos, "discarded non-Unit value")
-            }
+            if (tree.tpe.termSymbol == UnitClass.companionModule)
```

Comment:
You could probably add `UnitMod` and friends to `definitions` if you wanted.

Reply:
Will do!